package.json

```diff
@@ -34,7 +34,8 @@
     "release": "release-it",
     "example": "yarn --cwd example",
     "bootstrap": "yarn example && yarn install",
-    "code-review": "yarn lint && yarn typecheck && yarn test"
+    "code-review": "yarn lint && yarn typecheck && yarn test",
+    "generate-wallet-provider-client": "cd openapi && typed-openapi wallet-provider.yaml -o ../src/clients/generated/wallet-provider.ts -r zod && cd ../"
```

Comment:
`"typed-openapi openapi/wallet-provider.yaml -o src/client/generated/wallet-provider.ts -r zod"` ?

Reply:
note that `typed-openapi` types are pretty naive (it doesn't support dates, pattern strings, binary, ...). Moreover the community is not that big and latest commit was 3 moths ago. I don't know where this code will be used, but you may consider to use [openapi-zod-client](https://github.com/astahmer/openapi-zod-client) instead, which is pretty similar but has a better community, looks maintained and supports complex types.

Reply:
Unfortunately `openapi-zod-client `uses `axios` internally via the `zodios` client API and it is not possible to change the fetch function programmatically. It is not possible to use it here as the react-native based library is used by app IO that **do not have a node instance** to make `zodios` work and it is also necessary that the user of this library can declare the fetch function via a context to use. 

To overcome these limitations `openapi-zod-client` itself recommends using `typed-openapi`.

Since the open API of the entire wallet is always very simple, these are strings containing JWT encoded in base64, I believe that this library is sufficient. Do you have any other generator in mind that could be right for us?

Reply:
I've assessed sever generators and found no one that totally suits our needs. Suprisingly [our homemade generator](https://github.com/pagopa/openapi-codegen-ts), the one that io-app already uses for all APIs, is well positioned, but requires to relax the `zod` usage constraint since it's `io-ts` based.

Reply:
😞 `io-react-native-wallet` does not use **fp-ts** in favor of pure **typescript**. I would say to leave this and then in the future evaluate whether it is worth transforming our homemade generator for zod too.